### PR TITLE
added sample service test class

### PR DESCRIPTION
**Pull request by josephbate**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design
      documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary

- made changes to the sampleHumanDaoimpl to take in the id as a String which is in line with the data base
- created the sam pleServiceTest class to hundle all intergrated test 

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other
however there is a challenge i faced in `getSamplesForPatient_shouldReturnSamplesForPatient() `  where  database update operation violates foreign key constraints in both patient and SampleHuman tables which triggers the  a PersistenceException is occurring due to a ConstraintViolationException.
could you please give me any Suggestion
here's link tothe error: https://pastebin.com/w0DDeLGD
Cc. @mozzy11 
[Add any additional information or notes here]

### CODE REVIEW

**Comment by mozzy11** [comment]
@josephbate can you run `mvn spotless:apply` . the CI build fails because of formating error

**Comment by josephbate** [comment]
> @josephbate can you run `mvn spotless:apply` . the CI build fails because of formating error

Oh sorry 
Let me fixed that quickly

File: src/main/java/org/openelisglobal/samplehuman/daoimpl/SampleHumanDAOImpl.java [hunk 1]
```diff
@@ -145,7 +145,7 @@ public List<Sample> getSamplesForPatient(String patientID) throws LIMSRuntimeExc
                     + " sampleHuman.sampleId = sample.id and sampleHuman.patientId = :patientId order by"
                     + " sample.id";
             Query<Sample> query = entityManager.unwrap(Session.class).createQuery(sql, Sample.class);
-            query.setParameter("patientId", Integer.parseInt(patientID));
```
Comment (mozzy11) [inline src/main/java/org/openelisglobal/samplehuman/daoimpl/SampleHumanDAOImpl.java:148]: `sampleHuman.patientId` is stored as an Integer in the DB . so revert this comment

Comment (josephbate) [reply src/main/java/org/openelisglobal/samplehuman/daoimpl/SampleHumanDAOImpl.java:148]: But I was getting an error indicating otherwise in my test

Comment (josephbate) [reply src/main/java/org/openelisglobal/samplehuman/daoimpl/SampleHumanDAOImpl.java:148]: Anyways let me me revert so that we can see

**Comment by josephbate** [comment]
hello moses think i have fixed it

**Comment by josephbate** [comment]
ihave fixed it  
please review it again 
and also please adress my issue on the test case getSamplesForPatient_shouldReturnSamplesForPatient()
which i commented out

**Review by mozzy11 — APPROVED** [review]

**Pull request merged**